nestkernel/connection_manager.cpp

```diff
@@ -159,6 +162,13 @@ nest::ConnectionManager::set_status( const DictionaryDatum& d )
       "If structural plasticity is enabled, sort_connections_by_source can not "
       "be set to false." );
   }
+
+  updateValue< bool >( d, names::use_compressed_spikes, use_compressed_spikes_ );
+  if ( use_compressed_spikes_ and not sort_connections_by_source_ )
+  {
+    throw KernelException( "Can not use compressed spikes without sorting connections by source." );
```

Comment:
I would suggest to formulate this in a positive way, e.g., "Spike compression requires connections to be sorted  by source.".